include/pybind11/stl.h

```diff
@@ -420,13 +451,35 @@ struct array_caster {
         return l.release();
     }
 
-    PYBIND11_TYPE_CASTER_RVPP(ArrayType,
-                              const_name<Resizable>(const_name(""), const_name("Annotated["))
-                                  + const_name("list[") + value_conv::name + const_name("]")
-                                  + const_name<Resizable>(const_name(""),
-                                                          const_name(", FixedSize(")
-                                                              + const_name<Size>()
-                                                              + const_name(")]")));
+    // Code copied from PYBIND11_TYPE_CASTER macro.
+    template <typename T_, enable_if_t<std::is_same<ArrayType, remove_cv_t<T_>>::value, int> = 0>
+    static handle cast(T_ *src, const return_value_policy_pack &policy, handle parent) {
+        if (!src) {
+            return none().release();
+        }
+        if (policy == return_value_policy::take_ownership) {
+            auto h = cast(std::move(*src), policy, parent);
+            delete src; // WARNING: Assumes `src` was allocated with `new`.
```

Comment:
Do you know typically from whom is src allocated? If so, might be helpful to mention that here for more context.

Reply:
Or is this a typical requirement for any take_ownership policy on casters with pointer type src?

Reply:
I also added a source code comment here:

```
// Intentionally preserving the behavior exactly as it was before google/pywrapcc#30034
```

> Do you know typically from whom is src allocated?

No! :-)

I think this is one of the dark corners of pybind11, specifically the `PYBIND11_TYPE_CASTER` macro, which is used a lot. But ecosystem has grown around it ... better don't touch.